### PR TITLE
Add rust-src

### PR DESCRIPTION
**Pull request by Hofer-Julian**
This is especially interesting for IDE support with rust-analyzer

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
bot, build

**Comment by Hofer-Julian** [comment]
bot, build

**Comment by Hofer-Julian** [comment]
Rust-analyzer works fine now.
Not sure if dependency updates could be updated similar to what is already done to the rust binaries

File: org.freedesktop.Sdk.Extension.rust-stable.json [hunk 1]
```diff
@@ -123,6 +123,25 @@
                 "test -n \"$COMPAT_TARGET\" && cd \"rust-$COMPAT_TARGET\" && ./install.sh --prefix=/usr/lib/sdk/rust-stable --disable-ldconfig --verbose --components=rust-std-$COMPAT_TARGET,rust-analysis-$COMPAT_TARGET"
             ]
         },
+        {
+            "name": "rust-src",
```
Comment (alatiera) [inline org.freedesktop.Sdk.Extension.rust-stable.json]: weird this isn't provided by the binary distribution

Comment (Hofer-Julian) [reply org.freedesktop.Sdk.Extension.rust-stable.json]: True...

**Comment by alatiera** [comment]
what, you stupid github

**Comment by alatiera** [comment]
oh my god, I can't reopen this cause there are no commits *and* lost access to the PR branch. I SWEAR GIT IS THE MOST STUPID THING THAT WAS EVER INVENTED

**Comment by alatiera** [comment]
In other news was trying to rebase, apply and push the following patch

```
From 32632eb652a2e1308a93644609daa9ba5ed95c79 Mon Sep 17 00:00:00 2001
From: Julian Hofer <julianhofer@gnome.org>
Date: Sat, 19 Dec 2020 18:45:16 +0100
Subject: [PATCH] Add rust-src

This is especially interesting for IDE support with rust-analyzer
---
 ...freedesktop.Sdk.Extension.rust-stable.json | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git org.freedesktop.Sdk.Extension.rust-stable.json org.freedesktop.Sdk.Extension.rust-stable.json
index 4585588..d9dbcd9 100644
--- org.freedesktop.Sdk.Extension.rust-stable.json
+++ org.freedesktop.Sdk.Extension.rust-stable.json
@@ -123,6 +123,25 @@
                 "test -n \"$COMPAT_TARGET\" && cd \"rust-$COMPAT_TARGET\" && ./install.sh --prefix=/usr/lib/sdk/rust-stable --disable-ldconfig --verbose --components=rust-std-$COMPAT_TARGET,rust-analysis-$COMPAT_TARGET"
             ]
         },
+        {
+            "name": "rust-src",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/rust-lang/rust/archive/1.48.0.tar.gz",
+                    "sha256": "d5a72a5a375f147f90dd980d88b2625ed84f4944cab25ecf353da8f00d1c25c5",
+                    "x-checker-data": {
+                        "type": "anitya",
+                        "project-id": "7635",
+                        "url-template": "https://github.com/rust-lang/rust/archive/$version.tar.gz"
+                    }
+                }
+            ],
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p /usr/lib/sdk/rust-stable/lib/rustlib/src/rust/",
+                "cp -r library /usr/lib/sdk/rust-stable/lib/rustlib/src/rust/"]
+        }
         {
             "name": "scripts",
             "sources": [
-- 
2.29.2

```

**Pull request closed without merging**